### ISSUE TITLE
Sven Co-op map support

### ISSUE THREAD

**Issue by wootguy** [wootguy/SevenKewp#7]
A list of maps I've personally tested and confirmed are playable without major problems or missing features. Many maps are run through a script first to convert legacy entity logic and unsupported audio formats. I sometimes do a manual ripent for things that seem to be one-off bugs (e.g. trigger_push is stronger in this mod which can cause a softlock). A conversion script and my set of .ent files will come later. Possibly a separate map database as well.

Major hurdles for supporting recent maps:
- movement beyond the +/-4096 grid
- maps built with `-subdivide` above 240 cause a "bad surface extents" crash
- AngelScript support/replacement

I run a Half-Life server where you can try the mod and these maps:
207.148.12.159:27016

```
6doors
a_jungle_1 a_jungle_2 a_jungle_3 a_jungle_4 a_jungle_5 a_jungle_6 a_jungle_7
abandoned
actual actual2
arcad_v1 arcad2_v1
ascii_art2
assaultmesa2 assaultmesa2-2
auspices
beachxp
blackfacility
blackfacility2
blackmesaepf
blastcorpse1
bmtcoop
borgcube borgcube2
bridge2k
brokenpath
bunker
camp
captured
case1 case2 case3 case4 case5 case5a case6 case7 case8 case9
classic
clockwork
commap1 commap2 commap3 commap4 commap5 commap6 commap7
complex
conect
contact_demo
coop1 coop2 coop3
crisis1 crisis2
crystal crystal2
darkforest1 darkforest2
dd
deltar
deur
devious2_v2
dukemap2
dupa
durga
eastbase1
enc1 enc2 enc3
escape_series_1a escape_series_1b escape_series_2a escape_series_2b escape_series_2c escape_series_2d escape_series_2e escape_series_3a escape_series_3b escape_series_3c escape_series_3d escape_series_3e
execommand execommand2 execommand3
exhale1platinum exhale2platinum exhale3platinum exhale1 exhale4a exhale4b exhale5a exhale5b exhale6
extension extended
factions
farmhouse
fear
finalfight
flee
forest_clearing_1 forest_clearing_2 forest_clearing_3
fort_gorgon
fragers_rock
g-ara1 g-ara2 g-ara3 g-ara4 g-ara5
gash
gausslabbeta2
getout1
gf_greybox
ghost_buster
gitfest1g gitfest2g gitfest3g gitfest4g gitfest5g gitfest6g
gmansbuilding
gordonsci2
grunts grunts2
gruntwar
gruntwar2
gut_reaction gr_generate gr_lab gr_tunnel3 gr_corentry gr_coremain gr_drainage gr_coremain2 gr_reactors gr_finale
gv_mission1
halloween_cemetery
halloween_construction
haunted1 haunted2
haywire
hostage hostage2a hostage2b
hplanet
huntthehaunt
icestormx
infested infested2
infiltrate
inhuman
insertion2 insertion3 insertion4 insertion5
intruder
ioshockiron
jacked
jailbreak jailbreak2
justcause01 justcause02 justcause03
keenrace_beta19
kilionny
labra_x25keenb2
last last2 last3 last4
leprechaun1 leprechaun2 leprechaun3 leprechaun3-2
loz
mechanic01 mechanic02
mementocity
militarystrike_beta2
miseryb6
mountainpass
murks
music
nightmare-c1 nightmare-c2
nira1
nm_gruntwars_a nm_gruntwars_b nm_gruntwars_c nm_gruntwars_d
nm_helmsdeep
nm_moria
nm_uspninjas
nonsenselol_b1
nuearth
op4two1 op4two2
operationalpha operationalpha2 opalphacredits
opr1a1 opr1a2 opr1a3 opr1a4 opr1a5 opr1a6 opr1a7
osd_p01 osd_p02 osd_p03 osd_p04 osd_p05 osd_p06 osd_p07 osd_p08 osd_p09 osd_p10
osprey
otokotati_no_kouzan
peanyworldbeta1-1 peanyworldbeta1-2
phobia
prisonriot_erosion
projectg1 projectg2 projectg3 projectg4 projectg5 projectg6 projectg7 projectgscrt1a projectgscrt1b projectg8
psychic_agog_beta
puchi
purge
qe_house qe_sky1 qe_sky2 qe_sky3 qe_sky4
ragemap
rampage
re_evil_directors_cut
reco
rescue3
rhapsody_b1_1 rhapsody_b2_1 rhapsody3_beta1
road_to_shinnen
rooftopwar
rrrrrrrrr2
sc_activist sc_activist2 sc_activist3
sc_alley
sc_another
sc_army4final21
sc_avp sc_avp2 sc_avp3 sc_avp4 sc_avp5 sc_avp6 sc_avp7 sc_avp8 sc_avp9 sc_avp10_b2
sc_backattack
sc_campfire
sc_complex
sc_crazylab_final
sc_cyberspy
sc_dead_zone sc_dead_zone2 sc_dead_zone3_final sc_dead_zone4_final sc_dead_zone4-2_final
sc_deadthings_txt sc_deadthings2_txt sc_deadthings3_txt
sc_death_mission_txt sc_death_mission2_txt sc_death_mission3_txt sc_death_mission4_txt
sc_delta
sc_devian_legend
sc_doc
sc_duno_v3
sc_egypt
sc_er sc_er2
sc_evermore
sc_fortress
sc_frogger_final
sc_frontlinete1
sc_frostfire_beta1
sc_gosu
sc_greysnake1
sc_halo
sc_heist
sc_hidden
sc_intensity
sc_kill_rusty4
sc_lab
sc_lipas_1_sp
sc_mako
sc_marioland marioland2a marioland2b
sc_mazing
sc_mision73
sc_night
sc_nuclear_b01
sc_ooops
sc_opd
sc_operatives1
sc_persia
sc_phantasmish_beta
sc_picclub
sc_ratmaze_v4
sc_reflux
sc_residue
sc_severed
sc_silenthil_v1b
sc_snowadventure sc_snowadventure_2
sc_sonic_b2
sc_space2
sc_spaceviking1 sc_spaceviking2
sc_straightreloaded
sc_strangers sc_strangers2 sc_strangers3 sc_strangers4
sc_subsurface
sc_subwaytrain
sc_swars_installation sc_swars_installation1 sc_swars_installation2
sc_tetris1 sc_tetris2 sc_tetris3 sc_tetris4 sc_tetris5 sc_tetris6
sc_tomb
sc_tombofdeath_v12-1 sc_tombofdeath_v12-2 sc_tombofdeath_v12-3 sc_tombofdeath_v12-4 sc_tombofdeath_v12-5 sc_tombofdeath_v12-6
sc_toxic
sc_tropical1_final sc_tropical2_final sc_tropical3_final
sc_verc_18
sc_waterworks
sc_worlds
scirun
seaside1a seaside1b seaside1c
sectore
sektor_b_ma19
sence_bdayfun_b2
septic_morass_demo
sewer_systematic_f1 sewer_systematic_f2 sewer_systematic_f3
shanghai
shattered
sledge
sparrows_op_1
spb_act1 spb_act3
squad_gman1 squad_gman2
starcraft
starcraft2
storagearea2
strikeback1_beta3
subsurface
suicide_run
svencoop1 svencoop2
svencooprpg svencooprpg2 svencooprpg3
sweethomecity
syowa_japan
the_prison_v2
thebackdoor
theyrhungry theyrhungry2
toonrun1 toonrun2 toonrun3
tower1 tower2 tower3 tower4 tower5
ub_iseki1 ub_iseki2
ub_megaman ub_megaman2 ub_megaman3
ub_nagoya_v2
vger
virtualreality
warehouse-invasion
wired
wrongworld wrongworld2
yabma
zero
zomlab
```

**Comment by wootguy** [comment]
Current to-do list for maps I've tested and found don't work yet:
```
sc_tetrisfancv11:
- item_inventory

tele_down_b1:
- bad surface extents

schall_1:
- trigger_changevalue
- env_fog

swars_installation1:
- elevator only moves once

swars_installation1/2:
- ladders on vents don't work (stuck at top) (removed with ripent)

svenergy:
- water instakills you

sc_mountainre:
- w_flashlight.mdl freezes client, map worked otherwise

sc_strangers1/2:
- corpses have wrong pose (hd model)

sc_strangers3:
- sound precache overflow but barely

snarkcafe_bunker-beta2:
- garg gets kicked into air softlock, no bombs

snakes:
- intro broken? even enuze edit. softlock.

skylined:
- skill file (super wrench)

sc_tl_sky_fun_moon_final:
- high hp stuff
- pushables stuck

source_of_life:
- wrench
- skill file

shogo:
- skill file (super damage)

sc_tl_secrettask_1c:
- ladder toggle
- shorten cutscenes...

sc_s28:
- extreme lag

sc_russia:
- item_helmet
- item_armorvest
- env_fog

ruoste_demo:
- softlock if pressing button before using key

huntbgin_v2:
- bad surface extents
- gsr

reb:
- too many bsp models

sc_redfang:
- model precache overflow

ra_quad:
- too many bsp models

sc_rats_test2:
- toggle ladders

ragemap2012:
- changevalue/copyvalue/condition
- monster_sqnest
- env_sentence
- trigger_hurt_remote
- trigger_entity_iterator
- trigger_setcvar

quiet_neighborhood:
- changevalue/copyvalue/...

quarter:
- func_clip
- info_monster_goal
- monsters attacking breakables

prisoned-4:
- gate blocked by pushables(???)

sc_porton_betav1:
- item_inventory

little_mesa:
- nw_xeno.wad does not transfer despite being in .res file

septic_morass_demo:
- gonomes dont attack

escape_series_2a:
- box acid softlock
- crowbar is sometimes a knife model

wrongworld2:
- pendulum bridge got stuck (cant repro)

sc_outpost_v2_2:
- extreme lag

pagoda
- trigger_changevalue
- trigger_entity_iterator
- marksurfaces overflow(?)

escape_series_2a:
- box bridge, other box can float away softlock

polar_rescue (4.5):
- func_clip
- trigger_condition
- trigger_hurt_remote
- trigger_changevalue
- env_camera (?)

planet3c_r:
- item_inventory
- trigger_vote

pointless_b2:
- trigger_createentity
- func_clip

pizza_ya_san1:
- trigger_changevalue
- trigger_vote
- replace shotgun/soflam

phoenixwalkeralpha13:
- env_spritetrain

hlplu_1:
- trigger_changevalue
- trigger_copyvalue
- trigger_changesky
- trigger_condition
- trigger_createentity
- info_tele_destination
- sz_getspace crash

ph_iv1:
- trigger_changevalue
- item_inventory
- trigger_setcvar
- trigger_hurt_remote

op_blackmesa4:
- toggle ladders start off

osp2:
- otis can't route to door
- shorten osp1 train door openening grunt counter

devious2:
- overlapping cafeteria music

commap5:
- broken movement (can't repro)

commap7:
- barrel cant get out of door (it wont open all the way) (can't repro)

con3_huba:
- hunk_allocname: failed
- bad surface extents

blackmesaepf:
- mission 3 grunt gets stuck/runs away softlock

svencoop1:
- custom titles

oberon14:
- insane routing problem for medic, next node(?) across entire map (see wip map)

noob:
- trigger_changevalue
- env_spritetrain

it_has_leaks:
- trigger_changevalue

sc_obstangle:
- env_spritetrain

sc_tl_cf_library:
- trigger_changevalue

eqclassic:
- trigger_changemodel
- trigger_hurt_remote
- item_inventory
- trigger_vote
- env_sentence

lost_resource2:
- trigger_push not affecting stationary monster
- ripent the barney trigger in part1
- reduce health of boss stuff and apache in part2
- model crash

never:
- skill file

lostmaze:
- model precache overflow

lostone:
- model precache overflow

meatworld:
- bad surface extents

mogul:
- crashes
- env_fog

little_mesa:
- op4mortar

mistake:
- global sound replacement

sc_mirage1
- alloc block full

jbase1:
- extreme lag

jimblox_obby:
- bad surface extents

just_mansion:
- weapon_custom/angelscript/ripent

just_islave:
- weapon_custom/angelscript
- trigger_changevalue

gut_reaction:
- custom titles

hauntedsouls_a1:
- bad surface extents

high_rise_final1:
- model precache overflow (barely)

hidoi_map1
- Ladder trigger on/off

sc_hammerhead4:
- sci thinks you hurt him when blowing up mines, wont follow, softlock
- npkill 2 works?
- can't walk across gaps

hellmouth:
- monsters can't walk across gaps

nm_darkisland:
- pipewrench

exposedb1:
- pushable touch broken (slides off moving container)

fired_01:
- func_clip
- path_patrol

sc_face:
- op4mortar
- func_op4mortarcontroller

funny_apple_hunt:
- bad surface extents

g-ara4:
- zombie door doesn't open (game_counter works triggered manually but not with zombies?)

faraon1:
- func_clip

dread:
- kingpin
- trigger_changevalue

evilmansion:
- too many BSP models

devious_trials1
- can run with box lifted from underneath
- moving things rub the pushable and move it off the plat

sc_death_mission2_txt:
- keycard invisible because spawned inside entity

croodcoop:
- more button sounds

bm_nightmare_d_final:
- boxes get moved by elevator (touch needs to filter other pushables or smth)

it_has_leaks:
- trigger_changevalue

crossed_paths:
- angelscript

sc_cyclone:
- bad surface extents

airport_beta:
- trigger_condition
- env_spritetrain

keenrace_2017:
- trigger_changevalue
- way too many models

fire_temple:
- pipewrench
- trigger on arrival (info_teleport_destination)

sc_worms_series:
- bad surface extents

sc_darknebula:
- model precache overflow

5am:
- trigger_copyvalue
- trigger_changevalue
- trigger_createentity

desertbus_v1a:
- bad surface extents

legends_p01:
- overflowed clipnodes!?

afrikakorps:
- game_player_counter
- trigger_changevalue

bigcave:
- alloc block full

arcticmission:
- map bigger than +/-4096

sc_cyclone:
- bad surface extents

shockraid_jungle:
- tank can't rotate
- drone tanks are bad too

desertcircle:
- op4mortar
- func_op4mortarcontroller
- env_fog
- trigger_entity_iterator
- trigger_changevalue
- trigger_hurt_remote

central_bunker_r6
- trigger_condition
- env_spritetrain
- trigger_changevalue

sc_psyko:
- ld models
- stuckabat (alien controller jams the pig door)

loz1_forest:
- env_fog
- trigger_changevalue
- env_spritetrain
- trigger_condition
- trigger_copyvalue
- trigger_changevalue
- trigger_entity_iterator
- trigger_vote
- trigger_hurt_remote
- trigger_change_class
- trigger_createentity

uni_v1:
- trigger_changevalue
- trigger_hurt_remote
- _loop and _once behavior?

deluge:
- script replace model ogg
- global sound replacement
- custom door/train sounds
- trigger_changevalue
- trigger_vote

mystic_radar:
- env_fog

kyper_kuutio_2:
- trigger_changevalue

reco:
- stuck on verifying resources

infested:
- friendly osprey at end dropped enemies

forest_clearing_1:
- path_waypoint

sc_royals:
- trigger_changevalue
- func_clip

sc_robination:
- trigger_changevalue
- condition
- script
- entity iterator

island:
- trigger_copyvalue
- trigger_changevalue
- func_clip

bm64:
- wrench
- instant break wrench
- breakable spawn wrench
- op4mortar
- stukabat

keen halloween:
- env_spritetrain
- trigger_changevalue
- trigger_copyvalue
- trigger_createentity
- trigger_condition
- entity_iterator
- env_spritetrain

sc_confuse:
- game_player_counter

leprechaun:
- weapon pickup trigger (map 2)

arcad 1:
- can lift pushable through door (arcad)
- music doesn't loop (arcad)

sc_conspiracy:
- otis trouble going through door
```

**Comment by wootguy** [comment]
closing this in favor of Wayne's new spreadsheet
https://docs.google.com/spreadsheets/d/1UAEJQot_KHDnJj9jK1TRHbcBom1iwJ66A0-c-2RXKyw
